src/nupic/math/ArrayAlgo.hpp

```diff
@@ -906,25 +906,27 @@ namespace nupic {
   template <typename It>
   inline bool is_sorted(It begin, It end, bool ascending =true, bool unique =true)
   {
-    for (It prev = begin, it = ++begin; it < end; ++it, ++prev)
+    if (begin < end) {
```

Comment:
This is the actual fix that makes sure that the begin iterator won't get incremented past end. The rest of the changes in this file are the result of the editor eliminating hanging whitespace.